src/main/java/eu/hbp/mip/controllers/PathologiesAPI.java

```
@@ -35,41 +33,90 @@ public class PathologiesAPI {
+            mipEngineCDEsMetadata = gson.fromJson(
+                    response.toString(),
+                    new TypeToken<HashMap<String, Map<String, MetadataHierarchyDTO.CommonDataElement>>>() {
+                    }.getType()
+            );
+        } catch (ConnectException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        } catch (IOException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        }
+
+        Map<String, List<PathologyDTO.PathologyDatasetDTO>> datasetsPerPathology = new HashMap<>();
+
+        mipEngineCDEsMetadata.forEach( (pathology, cdePerDataset) ->  {
+            List<PathologyDTO.PathologyDatasetDTO> pathologyDatasetDTOS = new ArrayList<>();
+            cdePerDataset.forEach((dataset, cde) ->  pathologyDatasetDTOS.add(new PathologyDTO.PathologyDatasetDTO(dataset, cde.getLabel())));
+            datasetsPerPathology.put(pathology, pathologyDatasetDTOS);
+        });
+
+
+        return datasetsPerPathology;
+    }
+
+    public Map<String, MIPEngineAttributesDTO> getMIPEnginePathologyAttributes(Logger logger) {
+        Map<String, MIPEngineAttributesDTO> mipEnginePathologyAttributes;
+        // Get MIPEngine algorithms
+        try {
+            StringBuilder response = new StringBuilder();
+            HTTPUtil.sendGet(mipengineAttributesUrl, response);
+            mipEnginePathologyAttributes = gson.fromJson(
+                    response.toString(),
+                    new TypeToken<HashMap<String, MIPEngineAttributesDTO>>() {
+                    }.getType()
+            );
+        } catch (ConnectException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        } catch (IOException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        }
```

Catch all exceptions here, logging the exception and returning null.